src/basic_crawler.js

```diff
@@ -310,8 +310,8 @@ class BasicCrawler {
         // If we got here, it means we actually want to handle the error.
         request.pushErrorMessage(error);
 
-        // Reclaim and retry request if retryCount is not exceeded.
-        if (request.retryCount < this.maxRequestRetries) {
+        // Reclaim and retry request if request is not skipped and retryCount is not exceeded.
+        if (!request.skipped && request.retryCount < this.maxRequestRetries) {
```

Comment:
`request.skipped` kind of collides with `request.ignoreErrors` as the difference is that for `this.handleFailedRequestFunction()` is called for skipped but not for ignored one. We should unify this or rename one or both to make it clear.